src/routers/alpha-router/alpha-router.ts

```diff
@@ -542,4 +564,38 @@ export class AlphaRouter implements IRouter<AlphaRouterConfig>, ISwapToRatio<Alp
       'Log for gas model'
     );
   }
+
+  private calculateOptimalRatio(position: Position, sqrtRatioX96: JSBI, zeroForOne: boolean = true): Fraction {
+    const precision = JSBI.BigInt('1' + '0'.repeat(50))
+
+    const token0Proportion = SqrtPriceMath.getAmount0Delta(
+      sqrtRatioX96,
+      TickMath.getSqrtRatioAtTick(position.tickUpper),
+      precision,
+      true
+    )
+    const token1Proportion = SqrtPriceMath.getAmount1Delta(
+      sqrtRatioX96,
+      TickMath.getSqrtRatioAtTick(position.tickLower),
+      precision,
+      true
+    )
+
+    console.log('token0Delta', token0Proportion.toString())
+    console.log('token1Delta', token1Proportion.toString())
+
+    let optimalRatio =  new Fraction(
+      token0Proportion,
+      token1Proportion
+    )
+    if (!zeroForOne) optimalRatio = optimalRatio.invert()
+    return optimalRatio
+  }
+
+  private absoluteValue(fraction: Fraction): Fraction {
```

Comment:
could something like this live in the `Fraction` sdk module??

Reply:
yeah definitely, though wouldn't consider it blocking

Reply:
cool, will probably make a PR adding a few functions to the Fraction lib that would've helped me.